File: src-backend/routes/templates.js
const express = require('express');
const { openDatabase, supabase, isSupabaseConfigured } = require('../db/client');
const { syncTemplates, getMetaConfig } = require('../services/meta-service');

const router = express.Router();

router.get('/', async (req, res) => {
  try {
    let templates = [];

    if (isSupabaseConfigured) {
      const { data, error } = await supabase
        .from('templates')
        .select('*')
        .order('name', { ascending: true });

      if (error) throw error;
      templates = data || [];
    } else {
      const db = openDatabase();
      try {
        templates = await db.all('SELECT * FROM templates ORDER BY name');
      } finally {
        await db.close();
      }
    }

    res.json({ templates });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

router.post('/sync', async (req, res) => {
  try {
    const config = await getMetaConfig();
    if (!config || !config.access_token || !config.waba_id) {
      return res.status(400).json({ error: 'Meta configuration not found' });
    }
    const templates = await syncTemplates(config);
    res.json({ templates });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;

File: src-backend/routes/templates.py
from fastapi import APIRouter, HTTPException, status
from db.client import db_client
from services.meta_service import get_meta_config, sync_templates

router = APIRouter(prefix="/api/templates", tags=["templates"])

@router.get("")
async def get_templates_endpoint():
    """
    Retorna todas las plantillas locales de mensajes.
    """
    try:
        templates = await db_client.all("SELECT * FROM templates ORDER BY name")
        return {"templates": templates}
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))

@router.post("/sync")
async def sync_templates_endpoint():
    """
    Sincroniza y descarga las plantillas directamente desde Meta.
    """
    try:
        config = await get_meta_config()
        if not config or not config.get("access_token") or not config.get("waba_id"):
            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Meta configuration not found or incomplete")
        templates = await sync_templates(config["access_token"], config["waba_id"])
        return {"templates": templates}
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
